### ISSUE TITLE
[BUG] KL散度计算错误 - 使用绝对值差异而非真正的KL散度

### ISSUE THREAD

**Issue by huthvincent** [huthvincent/KV-Cache-Compression-in-Reinforcement-Learning#1]
**Labels**: `bug`, `critical`, `loss-function`

### 问题描述
`SMD/src/shadow_distillation_loss.py` 中 Track-2 的 KL 蒸馏损失使用了绝对值差异而非真正的 KL 散度，这在数学上是不正确的。

### 当前代码
```python
# 文件: SMD/src/shadow_distillation_loss.py
# 行号: 约 170-175 行

kl_per_token = (dense_lp - shadow_target_lp).abs()
kl_distill_loss = sum_of_sample_mean(kl_per_token)
```

### 问题分析
- `|log π_dense - log π_sparse|` 不是 KL 散度
- 真正的 KL 散度: `D_KL(P||Q) = Σ P(x) * log(P(x)/Q(x))`
- 当前实现只是 log-prob 差异的绝对值

### 建议修复
```python
# 方案A: 使用真正的 Forward KL (推荐)
# D_KL(π_shadow || π_dense) = E_shadow[log π_shadow - log π_dense]
kl_per_token = shadow_target_lp - dense_lp  # shadow 视角的 KL
kl_distill_loss = sum_of_sample_mean(kl_per_token.clamp(min=0))  # 只惩罚正向偏离

# 方案B: 使用 Reverse KL
# D_KL(π_dense || π_shadow) = E_dense[log π_dense - log π_shadow]
kl_per_token = dense_lp - shadow_target_lp
kl_distill_loss = sum_of_sample_mean(torch.exp(dense_lp) * kl_per_token)
```

### 影响范围
- 影响所有使用 SMD 方法的训练
- 可能导致训练不稳定或收敛到次优解
